### ISSUE TITLE
how to calculate metrics

### ISSUE THREAD

**Issue by Yuyok** [lyf1212/COCO-LC#7]
Hi author, thanks for your great project, when trying to reproduce your code, I used the model weights you gave for inference in the COCO, but when calculating the metrics, I found that the results are quite different from the ones in the paper, is it convenient for you to disclose the methodology of metrics calculations or what other details are there in there that I'm missing?

**Comment by lyf1212** [comment]
I used `pytorch-fid` to calculate FID, and colorfulness metric following BigColor. Is there marginal difference occur on Extended-COCO dataset or all dataset? Have you correctly setup the evaluation dataset? The overall Extended-COCO dataset keeps 1976 images and corresponding text prompt.

Btw, as I clarify in readme, `Initial random noises count a lot`. I didn't use the fixed initial noises in this repo when I wrote the initial paper, so there might be a little difference.

If you have other problems, please provide more information about your workflow and metrics results. If you have solved this issue, please consider to close it. Thank you!

**Comment by Yuyok** [comment]
I used the images in the JSON files you give, one image has two or three prompts or just one, so I test total 3520 images on Extended COCO-stuff(only 1940 images actually). And I use SAM and Mask2former to generate the mask and edges. The result(alpha=0.9), Colorfulness was only 34.21, while GT images had a Colorfullness of 45. Δcolorfullness=13.87. The same issue occurred with FID=17.56, which has a very large difference from the results in the paper. I using the default parameters you provided. Can you provide more detailed information. Thank you.

**Comment by lyf1212** [comment]
The main problem should be the difference of benchmark datasets.

I used the latest benchmark dataset of Extended-COCO in [this paper](https://arxiv.org/abs/2503.14974), and it can reproduce the following results, which is approximately aligned with your results:

![Image](https://github.com/user-attachments/assets/d3b27939-1dad-4caa-a291-7d66d1b390c4)

It seems that your test dataset is the latest one (1946 images, 3520 image-text pairs). I highly recommend you to adopt this benchmark in [this survey paper](https://arxiv.org/abs/2503.14974) because it is more formal and reasonable than the original setting. 

Also, if you are in a urgent usage of quantitative evaluation, you can report your reproduced metrics.

If there is still other issues, feel free to contact me, or please close this issue. Thanks a lot!

**Comment by Yuyok** [comment]
I previously read this version of the [paper](https://github.com/lyf1212/COCO-LC/blob/main/COCO-LC_fix_typo.pdf), and didn't notice that there was another [version](https://github.com/lyf1212/COCO-LC/blob/main/COCO-LC_improved.pdf), so the results are very different. I get it now, thank you so much.

**Comment by Yuyok** [comment]
Hi!

I tested the metrics entirety (alpha=0.9) and compared them to [your survey paper,](https://arxiv.org/abs/2503.14974), where FID and CF are already similar to the latest results of your paper, but when compared PSNR in **Table 3. PSNR, SSIM and LPIPS results on language-based colorization task of Language-based Image Colorization: a Benchmark and Beyond (Supplementary Material)**, the value I calculated was `PSNR=21.04, SSIM=0.8705`, which is quite different from your result of `PSNR=24.26, SSIM=0.9264`. 

And the PSNR and SSIM for other author's methods vary widely too (PSNR of [L-CAD](https://arxiv.org/abs/2305.15217) in [your survey paper](https://arxiv.org/abs/2503.14974) is 27.41, but I calculated was only 21.25. 

I use the `scikit-image` package to calculate PSNR and SSIM, and use the default parameters, pretrained weights and datasets, it's not just your method that varies a lot, but other methods as well, are there any details I'm overlooking when calculating this metric?

**Comment by lyf1212** [comment]
Thank you for your valuable question! I also use `scikit-image` to calculate PSNR and SSIM. The discrepancy is huge, so there might be some errors.

Followings are some possible reasons and recommendation:
- **Luminance Channel Replacement.** As I introduced in Sec. 4.1 of [paper](https://github.com/lyf1212/COCO-LC/blob/main/COCO-LC_improved.pdf), I replace the L channel of colorized output with the one of grayscale input. I also apply such operation to other methods.
- **Different resolution.** I use `512 x 512` as default resolution. In my previous experiment, different resolution brings some discrepancy.
- **Generation Variance.** Multi-step diffusion-based methods struggle with unstable inference. I recommend you to check with Table.3 in [L-CAD_arxiv_paper](https://arxiv.org/pdf/2305.15217), where shows original metrics on ImageNet-val5k dataset calculated by their authors. You can see that my result (23.06, 0.8991) is also different with their result (24.47, 0.92). You can test on more samples, and observe the variance. And I also provide an brief analysis in Appendix-Sec.6  of [my survey paper](https://arxiv.org/abs/2503.14974).

If you have other issues about [my survey paper](https://arxiv.org/abs/2503.14974) which is under review currently, may you open issues in [its corresponding repo](https://github.com/lyf1212/Color-Turbo)? I think they are very useful to improve the final version of this survey paper. Thank you!

**Comment by lyf1212** [comment]
It seems no remaining discussions here, so I will close this issue later.

If any other questions, feel free to reopen it!😊
